presto-base-arrow-flight/src/main/java/com/facebook/plugin/arrow/ArrowPageUtils.java

```diff
@@ -0,0 +1,686 @@
+    }
+
+    public static Block buildBlockFromListVector(ListVector vector, Type type)
+    {
+        if (!(type instanceof ArrayType)) {
+            throw new IllegalArgumentException("Type must be an ArrayType for ListVector");
+        }
+
+        ArrayType arrayType = (ArrayType) type;
+        Type elementType = arrayType.getElementType();
+        BlockBuilder arrayBuilder = type.createBlockBuilder(null, vector.getValueCount());
+
+        for (int i = 0; i < vector.getValueCount(); i++) {
+            if (vector.isNull(i)) {
+                arrayBuilder.appendNull();
+            }
+            else {
+                BlockBuilder elementBuilder = arrayBuilder.beginBlockEntry();
+                UnionListReader reader = vector.getReader();
+                reader.setPosition(i);
+
+                while (reader.next()) {
+                    Object value = reader.readObject();
+                    if (value == null) {
+                        elementBuilder.appendNull();
+                    }
+                    else {
+                        appendValueToBuilder(elementType, elementBuilder, value);
+                    }
+                }
+                arrayBuilder.closeEntry();
+            }
+        }
+        return arrayBuilder.build();
+    }
+
+    private static void appendValueToBuilder(Type type, BlockBuilder builder, Object value)
+    {
+        if (value == null) {
+            builder.appendNull();
+            return;
+        }
+
+        if (type instanceof VarcharType) {
+            // Convert value to string and write as Varchar
+            Slice slice = Slices.utf8Slice(value.toString());
+            type.writeSlice(builder, slice);
+        }
+        else if (type instanceof BigintType) {
+            if (value instanceof Long) {
+                type.writeLong(builder, (Long) value);
+            }
+            else if (value instanceof Integer) {
+                type.writeLong(builder, ((Integer) value).longValue());
+            }
+            else if (value instanceof JsonStringArrayList) {
+                JsonStringArrayList list = (JsonStringArrayList) value;
+                for (Object obj : list) {
+                    try {
+                        long longValue = Long.parseLong(obj.toString());
+                        type.writeLong(builder, longValue);
+                    }
+                    catch (NumberFormatException e) {
+                        throw new IllegalArgumentException("Invalid number format in JsonStringArrayList: " + obj, e);
+                    }
+                }
+            }
+            else {
+                throw new IllegalArgumentException("Unsupported type for BigintType: " + value.getClass());
+            }
+        }
+        else if (type instanceof IntegerType) {
+            if (value instanceof Integer) {
+                type.writeLong(builder, (Integer) value);
+            }
+            else if (value instanceof JsonStringArrayList) {
```

Comment:
won't `JsonStringArrayList` occur outside of ListVectors? Shouldn't we handle that case?

Reply:
I guess this is needed only for List implemenation may be not required for other primitive vectors